### PR TITLE
Added support for bitsandbytes quantization of models

### PR DESCRIPTION
**Pull request by AyushSawant18588**
Updates made:
Added bitsandbytes quantization of precision 4 and 8 bits. (precision bits can be passed though '-q' argument in llm/run.sh)

### CODE REVIEW

**Review by gavrishp — APPROVED** [review]

File: llm/utils/tsutils.py [hunk 1]
```diff
@@ -198,6 +199,22 @@ def set_model_params(model_name: str) -> None:
...
+    This function reads the precision to which the model weights are to be
+    quantized and sets it as environment variable for the handler
+    to read.
+
+    Args:
+        quantize_bits (int): BitsAndBytes Quantization Precision.
+    """
+    if quantize_bits not in [4, 8, 16]:
+        print("## Quantization precision bits should be either 4, 8 or 16")
+        sys.exit(1)
```
Comment (gavrishp) [inline llm/utils/tsutils.py:213]: fail if not using GPU?

Comment (AyushSawant18588) [reply llm/utils/tsutils.py:213]: oh yes will add that check

File: llm/handler.py [hunk 2]
```diff
@@ -104,8 +104,29 @@ def initialize(self, context):
         self.tokenizer.padding_side = "left"
         logger.info("Tokenizer loaded successfully")
 
+        quantize_bits = self.get_env_value("NAI_QUANTIZATION")
```
Comment (gavrishp) [inline llm/handler.py:107]: keep it simple here.  you are doing the check, just to type cast

```
if self.get_env_value("NAI_QUANTIZATION"):
   quantize_bits = int(self.get_env_value("NAI_QUANTIZATION"))
```

File: llm/utils/tsutils.py [hunk 3]
```diff
@@ -198,6 +200,25 @@ def set_model_params(model_name: str) -> None:
...
+    """
+    This function reads the precision to which the model weights are to be
+    quantized and sets it as environment variable for the handler
+    to read.
+
+    Args:
+        quantize_bits (int): BitsAndBytes Quantization Precision.
+    """
+    if quantize_bits and int(quantize_bits) not in [4, 8]:
+        print("## Quantization precision bits should be either 4 or 8")
```
Comment (gavrishp) [inline llm/utils/tsutils.py:213]: include message that default precision value is 16

**Pull request merged**